src/closure.md

```diff
@@ -0,0 +1,113 @@
+# Closure Expansion in rustc
+
+Let's start with an example
+
+### Example 1
+```rust
+fn foo<F>(closure: F)
+  where F: Fn() {
+  closure()
+}
+fn main() {
+  let x = 10;
+  foo(|| println!("Hi {}", x))
+}
+```
+If you compile Example 1 using the command
```

Comment:
"To see how the closure gets desugared, we'll take a look at the MIR that gets generated. To generate the MIR, we need to run `rustc` with the `-Zdump-mir=all` argument, like so:"

(We can link the MIR references to the MIR chapter; not sure if there is a chapter that describes dump-mir...)